libraries/AP_Filesystem/AP_Filesystem_ROMFS.cpp

```diff
@@ -62,6 +60,8 @@ int AP_Filesystem_ROMFS::close(int fd)
         return -1;
     }
     AP_ROMFS::free(file[fd].data);
+
+    WITH_SEMAPHORE(record_sem); // release file record
```

Comment:
should be above the free() I think

Reply:
Why? Nobody will re-use the record until the data pointer is nullptr.

This patch doesn’t try to protect using the same fd from multiple threads.

Reply:
it needs to either go above the free or can be removed completely. Right now it is protecting a store of a 32 bit value, which is atomic anyway (we assume 32 bit atomic in ArduPilot). So if this is needed at all then it is needed above the free

Reply:
Fair enough. I really hate that assumption, it is not at all true at the compiler level. I will move it.